apps/aecore/src/aec_chain.erl

```diff
@@ -112,15 +113,20 @@ start_link(GenesisBlock) ->
 stop() ->
     gen_server:stop(?SERVER).
 
-%% Returns the highest known block in the chain with its state trees.
+%% Returns the highest known block in the chain
 %%
 %% The highest known block may be lower than the highest block header
 %% in the chain as returned by `top_header/0`.
+%%
+%% It may happen that we skipped a block and top doesn't hold state trees
```

Comment:
@sennui as `top_block/0` does not return state trees, this version of `top/0` never returns state trees. I suggest to delete `top_with_state/0`, and replace `top/0` with the implementation of `top_with_state/0`.
(And refine doc of `top_block` and `top` if needed for clarification.)